tests/pcap/src/client_hello.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::handshake_message::Builder as MessageBuilder;
+use crate::handshake_message::Message;
+use anyhow::*;
+use std::option::Option;
+
+use crate::capture::*;
+
+#[derive(Debug, Clone, Default)]
+pub struct ClientHello {
+    pub message: Message,
+    pub ja3_hash: Option<String>,
+    pub ja3_str: Option<String>,
+}
+
+impl ClientHello {
+    const JA3_HASH: &'static str = "tls.handshake.ja3";
+    const JA3_STR: &'static str = "tls.handshake.ja3_full";
+
+    fn from_message(message: Message) -> Self {
+        let packet = &message.packet;
+        let ja3_hash = get_metadata(packet, Self::JA3_HASH).map(str::to_string);
+        let ja3_str = get_metadata(packet, Self::JA3_STR).map(str::to_string);
+        Self {
+            message,
+            ja3_hash,
+            ja3_str,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Builder(MessageBuilder);
+
+impl Builder {
+    pub fn inner(&mut self) -> &mut MessageBuilder {
+        &mut self.0
+    }
+
+    pub fn build(mut self) -> Result<Vec<ClientHello>> {
```

Comment:
The builder abstraction feels a little odd here and in `message::Builder`
- Builder is just MessageBuilder in a different coat
- you have to deref to `inner` to do the configuration for `client_hello::Builder`.
- each Builder returns `Vec`s of items, rather than a single item

Is there some other structure that might work better? I wonder if an abstraction around the actual Pcap would be useful.

```
struct Pcap {}

impl Pcap {
    fn get_messages(message_type: MessageType) -> Vec<Message>
    fn get_client_hellos() -> Vec<ClientHello>
}
```